File: sol/1436.py
from collections import defaultdict

class Solution:
    def destCity(self, paths: List[List[str]]) -> str:
        cities = set()
        outDeg = defaultdict(int)
        for a, b in paths:
            cities.update((a, b))
            outDeg[a] += 1
        return next(c for c in cities if outDeg[c] == 0)

File: sol/1436/1436.cpp
#include <unordered_set>

struct Solution {
    string destCity(vector<vector<string>>& paths) {
        unordered_set<string> s;
        for (auto &p : paths){
            s.insert(p[0]);
        }
        for (auto &p : paths){
            if (!s.contains(p[1])){
                return p[1];
            }
        }
        return "";
    }
};
